### PR TITLE
ci: use go-version-file and pin actions

### PR DESCRIPTION
**Pull request by tvoran**
Uses the `go-version-file` option in the setup-go action, instead of a separate job for getting the version from .go-version. Pins all github actions to SHAs (and upgrades a few). Bumps go to 1.19.6.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -14,25 +14,12 @@ env:
   PKG_NAME: "vault-k8s"
 
 jobs:
-  get-go-version:
```
Comment (tomhjp) [inline .github/workflows/build.yml:17]: Nice!

**Review by tomhjp — APPROVED** [review]
LGTM. Should we maybe add a dependabot config for github actions as well to help keep these up to date now that they're pinned?

**Review by swenson — APPROVED** [review]
LGTM

**Comment by tvoran** [comment]
@tomhjp Good idea, will add one in a follow-on PR.

**Pull request merged**